### PR TITLE
Query and update

### PR DESCRIPTION
**Pull request by AlyAtalla**
Add a column species of type string to your animals table. Modify your schema.sql file.
nside a transaction update the animals table by setting the species column to unspecified. Verify that change was made. Then roll back the change and verify that the species columns went back to the state before the transaction.
Inside a transaction:
Update the animals table by setting the species column to digimon for all animals that have a name ending in mon.
Update the animals table by setting the species column to pokemon for all animals that don't have species already set.
Verify that changes were made.
Commit the transaction.
Verify that changes persist after commit.
Write queries to answer the questions

### CODE REVIEW

**Review by TuyishimireEric — CHANGES_REQUESTED** [review]
# Status: REQUIRED CHANGES :recycle: 
### Hi  @AlyAtalla   👋 👋 
#### I'm @TuyishimireEric, Your code reviewer,
Good job so far 👍 
 there are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights 
✔ Professional PR
✔ correct sql commands

## Required Changes ♻️

- Check the comments below 👇

**Cheers and Happy coding!👏👏👏**

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by AlyAtalla** [comment]
Hi Eric, thank you for your review, I can't see your comment. if you mean about not opening a new PR, this is my first PR for this project. kindly any guidance would be appreciated.

File: schema.sql [hunk 1]
```diff
@@ -5,5 +5,6 @@ CREATE TABLE animals (
     date_of_birth DATE,
     escape_attempts INT,
     neutered BOOLEAN,
-    weight_kg DECIMAL(5, 2)
+    weight_kg DECIMAL(5, 2),
+    species VARCHAR(255)
```
Comment (mikethreels) [inline schema.sql]: - [x] Altering the table directly is not recommended, you should use the "ALTER TABLE" query in this case.

File: queries.sql [hunk 2]
```diff
@@ -16,4 +16,27 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+FROM animals
+GROUP BY neutered;
+
+-- What is the minimum and maximum weight of each type of animal?
+SELECT species, MIN(weight_kg) AS min_weight, MAX(weight_kg) AS max_weight
+FROM animals
+GROUP BY species;
+
+-- What is the average number of escape attempts per animal type of those born between 1990 and 2000?
+SELECT species, AVG(escape_attempts) AS avg_escape_attempts
```
Comment (mikethreels) [inline queries.sql]: - [x] good job writing the queries, however, it seems you accidentally didn't include the below requirements.

image

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Aly, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- good job adding data to the table

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql [hunk 3]
```diff
@@ -16,4 +16,61 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+WHERE date_of_birth BETWEEN '1990-01-01' AND '2000-12-31'
+GROUP BY species;
+
+-- Transaction 1: Update species to 'unspecified' and rollback
+BEGIN;
+UPDATE animals SET species = 'unspecified';
+-- Verify the change
+SELECT * FROM animals;
+-- Rollback
+ROLLBACK;
```
Comment (mikethreels) [inline queries.sql:50]: - [ ] according to the requirements, after rollback you should also verify if it changed back.

File: queries.sql [hunk 4]
```diff
@@ -16,4 +16,61 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+
+-- Transaction 2: Update species based on name and commit
+BEGIN;
+UPDATE animals SET species = 'digimon' WHERE name LIKE '%mon';
+UPDATE animals SET species = 'pokemon' WHERE species IS NULL;
+-- Verify the changes
+SELECT * FROM animals;
+-- Commit the transaction
+COMMIT;
```
Comment (mikethreels) [inline queries.sql:60]: - [ ] according to the requirements you should also verify if the changes persist after commit

File: queries.sql [hunk 5]
```diff
@@ -16,4 +16,61 @@ SELECT * FROM animals WHERE name != 'Gabumon';
...
+
+-- Transaction 3: Delete all records and rollback
+BEGIN;
+DELETE FROM animals;
+-- Verify if all records in the 'animals' table still exist (empty result set)
+SELECT * FROM animals;
+-- Rollback
+ROLLBACK;
+
```
Comment (mikethreels) [inline queries.sql]: - [ ] According to the requirements you should check if the records in the animals table still exist.

**Review by mikethreels — CHANGES_REQUESTED** [review]
Hi Aly, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
- Good job making the requested changes.

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @AlyAtalla , 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Good commit messages
✅ ... Good SQL queries

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**